uber/site_sections/budget.py

```diff
@@ -166,7 +166,15 @@ def generate_promo_codes(
                 params['code'] = code
                 promo_codes.append(PromoCode().apply(params))
 
+            message = check_all(promo_codes)
+            if message:
+                result['message'] = message
```

Comment:
Oh, this is a really unusual pattern for us. Normally we raise an HTTPRedirect with the message. Was there a specific reason you didn't do that, and do you think we should change our pattern for this?